### PR TITLE
Add public-scope

### PR DESCRIPTION
**Pull request by jackklpan**

### CODE REVIEW

**Comment by popcornylu** [comment]
I got this error message if no "access token" in cookie.
image

File: packages/admin-server/src/app.ts [hunk 1]
```diff
@@ -177,6 +189,108 @@ export const createApp = async (): Promise<{app: Koa, config: Config}> => {
...
+        if (sessionCached === false) {
+          console.log('Token: ', ctx.cookies.get('phapplication-session-id'));
+          return oidcCtrl.loggedIn(ctx, async () => {
+            if (scope === 'group') {
+              const groupAllowed = await checkUserGroup(ctx);
+              if (groupAllowed === false) {
+                throw Boom.forbidden('Request not authorized');
+              }
+            }
+            updateSessionToken(ctx);
```
Comment (popcornylu) [inline packages/admin-server/src/app.ts:269]: It seems that we don't reset the session timeout for each request? we can use https://github.com/node-cache/node-cache#change-ttl-ttl to update the ttl

**Review by popcornylu — COMMENTED** [review]
Only one cache ttl issue. Please take a look. Thanks

File: packages/admin-server/src/app.ts (hunk 1 above)
Comment (kentwelcome) [reply packages/admin-server/src/app.ts]: Fixed by adding new function call `updateSessionTTL` when `sessionToken` cache hit.

**Comment by popcornylu** [comment]
LGTM. But recommend to move the apps handlers to a seperate folder/file (like `oidc/*`) in next story and keep app.ts clean.

**Pull request merged**